src/platform/Ameba/BLEManagerImpl.cpp

```diff
@@ -810,6 +851,169 @@ bool BLEManagerImpl::IsSubscribed(uint16_t conId)
+    }
+
+    // Schedule DriveBLEState() to run.
+    PlatformMgr().ScheduleWork(DriveBLEState, 0);
+
+    return err;
+}
+
+void BLEManagerImpl::matter_blemgr_rx_char_write_cb(uint8_t conn_id, uint8_t * p_value, uint16_t len)
+{
+    sInstance.HandleRXCharWrite(p_value, len, conn_id);
+
+    PlatformMgr().ScheduleWork(DriveBLEState, 0);
+}
+
+void BLEManagerImpl::matter_blemgr_tx_char_cccd_write_cb(uint8_t conn_id, uint8_t indicationsEnabled, uint8_t notificationsEnabled)
+{
+    sInstance.HandleTXCharCCCDWrite((int) conn_id, (int) indicationsEnabled, (int) notificationsEnabled);
+
+    PlatformMgr().ScheduleWork(DriveBLEState, 0);
+}
+
+CHIP_ERROR BLEManagerImpl::matter_blemgr_tx_complete_cb(uint8_t conn_id)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    err = sInstance.HandleTXComplete((int) conn_id);
+
+    PlatformMgr().ScheduleWork(DriveBLEState, 0);
+
+    return err;
+}
+
+int BLEManagerImpl::matter_blemgr_callback_dispatcher(void * param, T_MATTER_BLEMGR_CALLBACK_TYPE cb_type, void * p_cb_data)
+{
+    BLEManagerImpl * blemgr = static_cast<BLEManagerImpl *>(param);
+
+    switch (cb_type)
+    {
+    case MATTER_BLEMGR_GAP_CONNECT_CB: {
+        T_MATTER_BLEMGR_GAP_CONNECT_CB_ARG * gap_connect_cb_arg = (T_MATTER_BLEMGR_GAP_CONNECT_CB_ARG *) p_cb_data;
+        blemgr->matter_blemgr_gap_connect_cb(gap_connect_cb_arg->conn_id);
+    }
+    break;
+    case MATTER_BLEMGR_GAP_DISCONNECT_CB: {
+        T_MATTER_BLEMGR_GAP_DISCONNECT_CB_ARG * gap_disconnect_cb_arg = (T_MATTER_BLEMGR_GAP_DISCONNECT_CB_ARG *) p_cb_data;
+        blemgr->matter_blemgr_gap_disconnect_cb(gap_disconnect_cb_arg->conn_id, gap_disconnect_cb_arg->disc_cause);
+    }
+    break;
+    case MATTER_BLEMGR_RX_CHAR_WRITE_CB: {
+        T_MATTER_BLEMGR_RX_CHAR_WRITE_CB_ARG * rx_char_write_cb_arg = (T_MATTER_BLEMGR_RX_CHAR_WRITE_CB_ARG *) p_cb_data;
+        blemgr->matter_blemgr_rx_char_write_cb(rx_char_write_cb_arg->conn_id, rx_char_write_cb_arg->p_value,
+                                               rx_char_write_cb_arg->len);
+    }
+    break;
+    case MATTER_BLEMGR_TX_CHAR_CCCD_WRITE_CB: {
+        T_MATTER_BLEMGR_TX_CHAR_CCCD_WRITE_CB_ARG * tx_char_cccd_write_cb_arg =
+            (T_MATTER_BLEMGR_TX_CHAR_CCCD_WRITE_CB_ARG *) p_cb_data;
+        blemgr->matter_blemgr_tx_char_cccd_write_cb(tx_char_cccd_write_cb_arg->conn_id,
+                                                    tx_char_cccd_write_cb_arg->indicationsEnabled,
+                                                    tx_char_cccd_write_cb_arg->notificationsEnabled);
+    }
+    break;
+    case MATTER_BLEMGR_TX_COMPLETE_CB: {
+        T_MATTER_BLEMGR_TX_COMPLETE_CB_ARG * tx_complete_cb_arg = (T_MATTER_BLEMGR_TX_COMPLETE_CB_ARG *) p_cb_data;
+        blemgr->matter_blemgr_tx_complete_cb(tx_complete_cb_arg->conn_id);
+    }
+    break;
+    default:
+        break;
+    }
+
+    return 0;
+}
+#else
+#if CHIP_ENABLE_ADDITIONAL_DATA_ADVERTISING
```

Comment:
could we use `#elif` ? given that the block of code is large, could we comment to what else branch this corresponds to (like `// not defined(CONFIG_MATTER_BLEMGR_ADAPTER) && CONFIG_MATTER_BLEMGR_ADAPTER`)?

Why is the additional data advertising depending on BLEMGR adapter not being used? would that option not apply to any type of BLE support?

Reply:
Fixed